spyderlib/widgets/sourcecode/codeeditor.py

```diff
@@ -2078,7 +2150,78 @@ def __in_block_comment(cursor):
             cursor3.setPosition(cursor.position())
             cursor3.select(QTextCursor.BlockUnderCursor)
             cursor3.removeSelectedText()
-        cursor3.endEditBlock()
+        cursor3.endEditBlock()  
+
+    #------Kill ring handlers
+    def kill_line_end(self):
+        """Kill the text on the current line from the cursor forward"""
+        cursor = self.textCursor()
+        cursor.clearSelection()
+        cursor.movePosition(QTextCursor.EndOfLine, QTextCursor.KeepAnchor)
+        if not cursor.hasSelection():

```

Comment:
I added a general note about where these functions came from in the section comment and addressed your grammar fixes.